### PR TITLE
Migrate to official MongoDB driver

### PR DESCRIPTION
**Pull request by alfrunes**
Migration to official mongodb driver from open-source version.

changelog: title

Signed-off-by: Alf-Rune Siqveland <alf.rune@northern.tech>

### CODE REVIEW

**Comment by mchalski** [comment]
@alfrunes 
good news:
- the migration logic seems ok, we don't 'change the past' as we agreed it's a priority
- the rest LGTM too; it's impossible to 100% review the logic, but it literally 'looks good', and if the tests run fine, we're good imo

bad news:
- 1 CR, a code organization level thing

**Comment by mchalski** [comment]
I'm approving but it would be useful for @kjaskiewiczz to have at least a brief look.

File: store/mongo/datastore_mongo.go [hunk 1]
```diff
@@ -288,15 +288,16 @@ func NewMongoClient(ctx context.Context, c config.Reader) (*mongo.Client, error)
 	wc.WithOptions(writeconcern.W(1), writeconcern.J(true))
 	clientOptions.SetWriteConcern(wc)
```
Comment (kjaskiewiczz) [inline store/mongo/datastore_mongo.go]: changes in this file shouldn't be a part of "vendor" commit

File: LIC_FILES_CHKSUM.sha256 [hunk 2]
```diff
@@ -49,13 +50,16 @@ a2e6011b982212228008c4386616a4818eb5fdf9ea5805b7a90500e6ce895aaf  vendor/github.
...
+9e203bd56d54ad11a8106e63e7a738363634f23ac91e14bddb2e35c5e7545a72  vendor/github.com/go-stack/stack/LICENSE.md
 #
 # LGPL3 with static linking exception.
 b40930bbcf80744c86c46a12bc9da056641d722716c378f5659b9e555ef833e1  vendor/gopkg.in/yaml.v2/LICENSE
-
+#
 # Addendum to the above.
 a94710b55e03b5285f77d048c5ba61bb9d6ee04a06c0eb90e68821e11b0c707a  vendor/gopkg.in/yaml.v2/LICENSE.libyaml
+# Copyright Go authors
+2d36597f7117c38b006835ae7f537487207d8ec407aa9d9980794b2030cbc067  vendor/golang.org/x/sync/LICENSE
```
Comment (kjaskiewiczz) [inline LIC_FILES_CHKSUM.sha256]: isn't this a BSD-3-Clause ?

File: LIC_FILES_CHKSUM.sha256 [hunk 3]
```diff
@@ -49,13 +50,16 @@ a2e6011b982212228008c4386616a4818eb5fdf9ea5805b7a90500e6ce895aaf  vendor/github.
...
+f69f157b0be75da373605dbc8bbf142e8924ee82d8f44f11bcaf351335bf98cf  vendor/github.com/golang/snappy/LICENSE
```
Comment (kjaskiewiczz) [inline LIC_FILES_CHKSUM.sha256]: same as above

File: LIC_FILES_CHKSUM.sha256 (hunk 2 above)
Comment (alfrunes) [reply LIC_FILES_CHKSUM.sha256]: TBH: This licensing stuff is a bit foreign to me. I'll change it asap.

File: store/mongo/datastore_mongo.go (hunk 1 above)
Comment (alfrunes) [reply store/mongo/datastore_mongo.go]: Well spotted! I'll move it back to where it belongs.

**Pull request merged**